modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -106,13 +106,26 @@ static GDScriptParser::DataType make_native_meta_type(const StringName &p_class_
 	return type;
 }
 
-static GDScriptParser::DataType make_native_enum_type(const StringName &p_native_class, const StringName &p_enum_name) {
+// In enum types, native_type is used to store the class (native or otherwise) that the enum belongs to.
+// This disambiguates between similarly named enums in subclasses or outer classes
+static GDScriptParser::DataType make_enum_type(const StringName &p_enum_name, const String &p_base_name = "", const bool p_meta = false) {
 	GDScriptParser::DataType type;
 	type.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT;
 	type.kind = GDScriptParser::DataType::ENUM;
-	type.builtin_type = Variant::INT;
+	type.builtin_type = p_meta ? Variant::DICTIONARY : Variant::INT;
+	type.enum_type = p_enum_name;
 	type.is_constant = true;
-	type.is_meta_type = true;
+	type.is_meta_type = p_meta;
+
+	if (!p_base_name.is_empty()) {
+		type.native_type = p_base_name + "::" + p_enum_name;
```

Comment:
This was chosen mostly for aesthetic reasons!

Some classes are defined by their filename, so when printing errors, you would end up with awkward enum names like `c:/my_local_hard_drive/test_project/my_class_gdscript_file.gd.MyEnum`. So instead, by using `::`, when printing errors you throw away the path, keep the file, and end up with `my_class_gdscript_file.gd::MyEnum`, which I thought was a lot more readable.

Technically speaking, for enums, `native_type` is only used to compare `DataTypes` and check if enum meta_types match. As long as they uniquely identify their enum, there shouldn't be any issues I believe!

Reply:
> so when printing errors, you would end up with awkward enum names like `c:/my_local_hard_drive/test_project/my_class_gdscript_file.gd.MyEnum`.

That's only the case for the tests. Otherwise, there will mostly always have the `res://` prefix to the GDScript files.

Reply:
@adamscott yes!!! Except that when you run your tests from the executable, it uses the entire path! So if my path is different than wherever GitHub runs Godot tests, it will fail. Probably to do with how the unit test framework loads GDScripts?